src/librustc_trans/trans/adt.rs

```diff
@@ -462,36 +503,38 @@ fn find_discr_field_candidate<'tcx>(tcx: &ty::ctxt<'tcx>,
             match field_ty.sty {
                 ty::TyRawPtr(ty::TypeAndMut { ty, .. }) if !type_is_sized(tcx, ty) => {
                     path.extend_from_slice(&[0, FAT_PTR_ADDR]);
-                    Some(path)
+                    Some((path, C_null(type_of::sizing_type_of(cx, ty))))
```

Comment:
`C_null(Type::i8p(cx))` here too.

Oh, and I said never used because all these cases with more than one entry in `path` end up as `StructWrappedNullablePointer` which doesn't actually use the forbidden value as far as I could tell?